hass_client/client.py

```diff
@@ -307,6 +343,18 @@ async def _process_messages(self) -> None:
                     break
 
                 if msg.type == WSMsgType.ERROR:
+                    # Home Assistant can produce some *very* large messages, and there's
+                    # no sign of a chunking API turning up soon. So check if we're losing
+                    # the connection due to message size.
+                    if msg.data.code == aiohttp.WSCloseCode.MESSAGE_TOO_BIG:
+                        # Parse the attempted size out, and schedule a reconnect.
+                        if (m := re.match(r"Message size (\d+)", msg.data.args[1])) is not None:
+                            attempted_message_size = int(m.group(1))
+                            # Set to 2x what they attempted to send us so hopefully we'll succeed
+                            # on reconnect.
+                            self._max_msg_size = attempted_message_size * 2
+                            raise ConnectionFailedDueToLargeMessage()
+
```

Comment:
I think you have a very specific usecase here, it is not very common to have THIS MANY entities.
So my suggestion would be to just increase the default message limit for now and keep it simple.

Reply:
The theory is that the user is already aware, or doesn't care - which for me *is* true because my HA installation operates just fine (hence how I discovered this issue - turning on iBeacons probably did it and I've not cleaned up yet).

So the logic is that if HA doesn't hit this (and it doesn't seem to have any limit other then browser memory) then the user probably doesn't care either for any practical case because this is HA-specific.

Basically: if the normal HA client isn't breaking (which is much heavier on memory) then the Python client shouldn't break either - it would Just Work(TM).

The other side of that is an obvious reason to use this library would be to do something like scripting a bulk clean up entities as well.

Reply:
Makes sense, although I'm a bit worried about the readability/maintainability of this client with this added overhead.
If you adjust the PR so its compatible with the latest refactor, I'm fine merging it.

Another, more simple approach could be to just have a config param to set the default message size for more advanced setups.